library/src/blas3/rocblas_trmm.hpp

```diff
@@ -0,0 +1,1488 @@
+                                }
+                            }
+                            //
+                            //                      C := T1', the transpose of T1 is copied back
+                            //                      to C.
+                            //
+                            for(int j = jj; j <= jj + jsec - 1; j++)
+                            {
+                                CHECK_INTERNAL_ROCBLAS_ERROR(
+                                    (rocblas_copy_template<NB, T>(handle,
+                                                                  isec,
+                                                                  &dt1[j - jj],
+                                                                  0,
+                                                                  rb,
+                                                                  0,
+                                                                  &c[ii - 1 + (j - 1) * ldc],
+                                                                  0,
+                                                                  1,
+                                                                  0,
+                                                                  1)));
+                            }
+                        }
+                        //
+                        //                  C := alpha*A*C + C, general matrix multiply
+                        //                  involving a rectangular block of A.
+                        //
+                        if(ii + isec <= m)
+                        {
+                            CHECK_INTERNAL_ROCBLAS_ERROR((rocblas_gemm_template<false, false>(
+                                handle,
+                                rocblas_operation_none,
+                                rocblas_operation_none,
+                                isec,
+                                n,
+                                m - ii - isec + 1,
+                                alpha,
+                                0,
+                                (const T*)(&a[ii - 1 + (ii + isec - 1) * lda]),
+                                0,
+                                lda,
+                                0,
+                                (const T*)(&c[ii + isec - 1]),
+                                0,
+                                ldc,
+                                0,
+                                &one,
+                                0,
+                                &c[ii - 1],
+                                0,
+                                ldc,
+                                0,
+                                1)));
+                        }
+                    }
+                }
+                else
+                {
+                    //
+                    //             Form  C := alpha*A'*C. Left, Upper, Transpose.
+                    //
+                    delta     = *alpha;
```

Comment:
Is `alpha` guaranteed on the host? 

The usual way to handle scalars is explained as #3 [on this page](https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/develop/.github/CONTRIBUTING.md)

Reply:
Removed use of delta.